drivers/ieee802154/Kconfig

```diff
@@ -80,10 +80,18 @@ source "drivers/ieee802154/Kconfig.dw1000"
 
 source "drivers/ieee802154/Kconfig.uart_pipe"
 
+# TODO: This option has been deprected and must neither be extended nor
+#       used by new/existing drivers. It is to be removed in the future.
+#       Features from any of the IEEE 802.15.4 standard versions should
+#       be enabled by fine-grained feature flags and not linked to any
+#       specific version of the standard.
 config IEEE802154_2015
-	bool "Support for IEEE 802.15.4-2015 frames"
+	bool "Support for OpenThread 1.3 IEEE 802.15.4 features (DEPRECATED)"
```

Comment:
Good point. I thought a bit about it and would like to suggest that I introduce this once I provide an alternative that users can use. For now, if you want to support auto-ack or Thread 1.2 you're actually obliged to use those options.

Is there a way to signal "deprecated for development" as opposed to "deprecated for usage"?

Reply:
No there isn't. So perhaps we should rather deprecate existing options only when we have an alternative to use?

Reply:
I agree - does it make sense to leave the TODO in there (which should be enough to alert developers) and remove the DEPRECATED word?

Reply:
It would make sense to me - we clearly signal the options will be removed in the future, but don't deprecate until we have an alternative to propose (or simply no longer use the option on our own).

Reply:
I updated the PR accordingly and removed the commit deprecating IEEE802154_2015 in favor of your upcoming PR that will completely remove the option.